### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by lihter**
Created custom search bar view `MovieSearchBar` and added it to the `HomePageTwoVC`.
The search bar has cancel button as well as delete button that makes the textfield empty

https://user-images.githubusercontent.com/82365531/126490870-aa620dae-91ba-4007-acba-0f660e453195.mp4

### CODE REVIEW

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        styleViews()
+        defineLayoutForViews()
+    }
+    
+    func createViews() {
+        searchGrayFieldView = UIView()
+        addSubview(searchGrayFieldView)
+        
+        let image = ImageEnum.searchBarIcon.image
+        searchBarIconImageView = UIImageView(image: image)
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:17]: You can just write `UIImageView(image: ImageEnum.searchBarIcon.image)`

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    
+    func styleViews() {
+        searchGrayFieldView.layer.cornerRadius = grayFieldCornerRadius
+        searchGrayFieldView.backgroundColor = .searchFieldGray
+        
+        searchBarIconImageView.contentMode = .scaleAspectFit
+        
+        searchTextField.attributedPlaceholder = NSAttributedString(
+            string: "Search",
+            attributes: [NSAttributedString.Key.foregroundColor: UIColor.searchPlaceholderColor,
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:41]: You should be able to write attributes keys without `NSAttributedString.Key` just `.foregroundColor`.

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                         NSAttributedString.Key.font: UIFont(name: Fonts.proximaMedium, size: 16) ?? .systemFont(ofSize: 16)])
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:42]: For fonts, you can create UIFont extension where you'll have sth like
```
public static let heading1: UIFont = .regular(size: 60)

...

static func regular(size: CGFloat = 12) -> UIFont {
  UIFont(with: .regular, size: size)
}
```

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        
+        deleteTextButton.snp.makeConstraints {
+            $0.trailing.equalToSuperview().inset(2 * offset)
+            $0.centerY.equalToSuperview()
+            $0.height.equalTo(deleteTextButtonHeight)
+        }
+    }
+    
+    func changeCancelButtonVisibility() {
+        if(cancelButton.isHidden) {
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:93]: No need for `(` and `)` in if statement.

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                self.searchGrayFieldView.snp.remakeConstraints {
+                    $0.leading.equalToSuperview().offset(4 * self.offset)
+                    $0.trailing.equalTo(self.cancelButton.snp.leading).offset(-4 * self.offset)
+                    $0.top.bottom.equalToSuperview()
+                }
+                self.deleteTextButton.isHidden.toggle()
+                self.searchGrayFieldView.superview?.layoutIfNeeded()
+            }, completion: { _ in
+                self.cancelButton.isHidden.toggle()
+            })
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:104]: This should be:
```swift
UIView.animate(
    withDuration: 0.5,
    animations: {
        // animations block
    },
    completions: {
        // completion block
    })
```

**Review by marinzg — CHANGES_REQUESTED** [review]

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:17]: I changed it [here](https://github.com/lihter/Movie-App/pull/24/files#diff-6a7e0c3cc088b3a980b8a92c08ad5d190f909d67f14aa13f150bdc992ab52302R16).

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift (hunk 2 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:41]: Changed [here](https://github.com/lihter/Movie-App/pull/24/files#diff-6a7e0c3cc088b3a980b8a92c08ad5d190f909d67f14aa13f150bdc992ab52302R40) and on every other occurrence in code.

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift (hunk 3 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:42]: I [created](https://github.com/lihter/Movie-App/pull/24/files#diff-019a1bcedcb8e120ecbea6c4a01e205958bbfa9c7e0a20366e002a99374c42e1R3) an extension as advised.

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift (hunk 4 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:93]: Removed `(` `)` from [that](https://github.com/lihter/Movie-App/pull/24/files#diff-6a7e0c3cc088b3a980b8a92c08ad5d190f909d67f14aa13f150bdc992ab52302R94) and every other if statement.

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift (hunk 5 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieSearchBar+Design.swift:104]: I [formated](https://github.com/lihter/Movie-App/pull/24/files#diff-6a7e0c3cc088b3a980b8a92c08ad5d190f909d67f14aa13f150bdc992ab52302R95) the code as advised.

**Review by marinzg — APPROVED** [review]

**Pull request merged**